src/lib/utils/fetch-cost-of-carry.ts

```diff
@@ -40,9 +48,87 @@ export async function fetchCostOfCarry(
       return
     }
 
-    setCostOfCarry(
-      Number(borrowedAsset.supplyAPY) - Number(borrowedAsset.variableBorrowAPY),
+    const collateralDebtTokens =
+      leverageCollateralDebt[inputOutputToken.arbitrumAddress!]
+
+    const collateralAmountPromise = publicClient.readContract({
+      address: collateralDebtTokens.collateralAmountToken as Address,
+      abi: ArbitrumLeverageTokenAbi,
+      functionName: 'balanceOf',
+      args: [inputOutputToken.arbitrumAddress as Address],
+    })
+    const collateralDecimalsPromise = publicClient.readContract({
+      address: collateralDebtTokens.collateralAmountToken as Address,
+      abi: ArbitrumLeverageTokenAbi,
+      functionName: 'decimals',
+    })
+    const debtAmountPromise = publicClient.readContract({
+      address: collateralDebtTokens.debtAmountToken as Address,
+      abi: ArbitrumLeverageTokenAbi,
+      functionName: 'balanceOf',
+      args: [inputOutputToken.arbitrumAddress as Address],
+    })
+    const debtDecimalsPromise = publicClient.readContract({
+      address: collateralDebtTokens.debtAmountToken as Address,
+      abi: ArbitrumLeverageTokenAbi,
+      functionName: 'decimals',
+    })
+
+    const [collateralAmount, collateralDecimals, debtAmount, debtDecimals] =
+      await Promise.all([
+        collateralAmountPromise,
+        collateralDecimalsPromise,
+        debtAmountPromise,
+        debtDecimalsPromise,
+      ])
+
+    const navProvider = new NavProvider()
+    const nav = await navProvider.getNavPrice(
+      inputOutputToken.symbol,
+      ARBITRUM.chainId,
     )
+
+    const collateralPrice = await fetchCoingeckoTokenPrice(
+      collateralDebtTokens.collateralPriceToken,
+      1,
+    )
+    const debtPrice = await fetchCoingeckoTokenPrice(
+      collateralDebtTokens.debtPriceToken,
+      1,
+    )
+
+    const supplyAPY = Number(borrowedAsset.supplyAPY)
+    const borrowAPY = Number(borrowedAsset.variableBorrowAPY)
+
+    const collateralCalculated =
+      Number(formatUnits(collateralAmount, collateralDecimals)) *
+      collateralPrice
+
+    const debtCalculated =
+      Number(formatUnits(debtAmount, debtDecimals)) * debtPrice
+
+    // TODO: Delete
+    console.log({
+      symbol: inputOutputToken.symbol,
+      collateralAmount,
+      collateralCalculated,
+      collateralAmountFormatted: Number(
+        formatUnits(collateralAmount, collateralDecimals),
+      ),
+      collateralPrice,
+      collateralDecimals,
+      debtAmount,
+      debtAmountFormatted: Number(formatUnits(debtAmount, debtDecimals)),
+      debtCalculated,
+      debtDecimals,
+      nav,
+      supplyAPY,
+      borrowAPY,
+    })
+
+    const costOfCarry =
+      (collateralCalculated * supplyAPY - debtCalculated * borrowAPY) / nav
+    setCostOfCarry(costOfCarry)
```

Comment:
Looking at the dune dash: https://dune.com/index_coop/indexcoop-perps-vs-leverage-tokens#cost-of-carry-leverage-tokens, it currently looks like what is shown for ETH2x (85.59%) could be closer to ETH3x but of by 10x? Maybe it's just coincidence.